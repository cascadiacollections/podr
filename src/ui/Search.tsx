import { h, JSX, FunctionComponent } from 'preact';

import { useCallback } from 'preact/hooks';
interface ISearchProps {
  onSearch: (query: string) => void;
}

export const Search: FunctionComponent<ISearchProps> = ({ onSearch }: ISearchProps) => {
  const onKeyDown = useCallback((event: JSX.TargetedKeyboardEvent<HTMLInputElement>) => {
    if (event.key === 'Enter') {
      const target: HTMLInputElement = event.target as HTMLInputElement;

      onSearch(target.value.trim());
    }
  }, [onSearch]);

  return <input class='form-control' type='search' placeholder='Search podcasts' onKeyDown={onKeyDown} />
}
