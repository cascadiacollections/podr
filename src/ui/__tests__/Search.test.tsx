import { h } from 'preact';
import { render } from '@testing-library/preact';
import { Search } from '../Search';
import { Signal } from '@preact/signals';

// Mock the hooks to prevent errors
jest.mock('../../utils/hooks', () => ({
  useAnalytics: () => ({
    trackSearch: jest.fn(),
  }),
  useClassNames: jest.fn((...args) => {
    // Simple mock that handles object conditions
    return args.flat().map(arg => {
      if (typeof arg === 'string') return arg;
      if (typeof arg === 'object' && arg !== null) {
        return Object.entries(arg)
          .filter(([, value]) => value)
          .map(([key]) => key)
          .join(' ');
      }
      return '';
    }).filter(Boolean).join(' ');
  }),
}));

describe('Search component', () => {
  test('renders correctly', () => {
    const onSearch = jest.fn();
    const { container } = render(<Search onSearch={onSearch} />);
    expect(container).toMatchSnapshot();
  });

  test('renders with query signal', () => {
    const onSearch = jest.fn();
    const querySignal = { value: 'test query' } as Signal<string>;
    const { container } = render(<Search onSearch={onSearch} querySignal={querySignal} />);
    expect(container).toMatchSnapshot();
  });
});