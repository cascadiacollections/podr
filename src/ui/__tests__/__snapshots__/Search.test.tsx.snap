// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders correctly 1`] = `
<div>
  <form
    role="search"
  >
    <input
      aria-label="Search podcasts"
      class="form-control"
      placeholder="Search podcasts"
      type="search"
    />
    <button
      aria-label="Submit search"
      class="search-button"
      type="submit"
    >
      Search
    </button>
  </form>
</div>
`;

exports[`Search component renders with query signal 1`] = `
<div>
  <form
    role="search"
  >
    <input
      aria-label="Search podcasts"
      class="form-control"
      placeholder="Search podcasts"
      type="search"
    />
    <button
      aria-label="Submit search"
      class="search-button"
      type="submit"
    >
      Search
    </button>
  </form>
</div>
`;
